### DISCUSSION TITLE
[Request] 更改“新建助手”“随便聊聊”逻辑

### DISCUSSION THREAD

**Discussion by samxiaowastaken** [lobehub/lobehub#1318] [General]
### 🥰 Feature Description

我用了一段时间的lobechat，发现这个“随便聊聊”和设置中的“默认助手”并不是同一个东西。

这是我所观察到的逻辑：
- ”随便聊聊“确实是lobechat自带的助手，继承设置中“默认助手”的prompt
- “默认助手”是新建助手时的模板
- 但是”随便聊聊“右上角的菜单会进入”默认助手“的设置（而不是和自定义助手的”会话设置“）
![image](https://github.com/lobehub/lobe-chat/assets/76235458/24167230-a466-4058-be14-353c37f9adb1)
- “默认助手"处设置助手名字和描述并不会影响”随便聊聊“
- 从其他助手的设置可以进入“随便聊聊”的真实设置界面“会话设置”，但是改名字和描述不会生效
![image](https://github.com/lobehub/lobe-chat/assets/76235458/7585aae1-b89d-4496-b97e-cc9942e7937e)
- 助手栏右上角的新建图标是新建助手（并且使用“默认助手”模板），而不是新建对话

这逻辑在使用过程中非常怪异，以至于我反复对“默认助手”进行设置，最后查看issue才发现之前有提到过类似的问题（#538）

我理解lobechat是以“助手-对话”这种逻辑存在的，但是我认为”随便聊聊“的设置逻辑混乱（以及不能改名字和描述）、并且”默认助手“的菜单名字不够易于理解。而且似乎新建对话这个功能比新建助手更加常用，但”新建助手“的按钮过于显眼，并且图标不够明确。

### 🧐 Proposed Solution

对于助手设置：
- ”随便聊聊“的右上角菜单应该同样导向“会话设置”
- “随便聊聊”的“会话设置”应该可以正常更改名字和描述，并且prompt和”默认助手“独立
- “默认助手”改为“新建助手模板”（或者其他更突出“模板”的名字）

对于“新建助手”或者“新建对话”：
- 设置中提供开关来关闭列表右上方的按钮，或者提供选项设置按钮为”新建助手“或者”用随便聊聊进行对话“
- 更改右上角新建助手的图标，更加突出“助手”而不是“对话”

其他易用性建议（这似乎应该是另外的功能建议）：
- 提供一个更强大的检索界面（因为“助手-对话”有些时候比较难查找）
- 提供另外一种界面逻辑选项，类似openAI的官网界面（陈列来自所有助手的所有对话，新建对话时选择助手）

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
感谢建议，先说下随便聊聊的助手设置 的发展过程：

随便聊聊作为LobeChat 内置的助手，我们在设计之初的目的有两个：

1. **快速引导用户上手**： 新用户一上来就能用，不用再自己创建一个默认助手；
2. **承接一些通用的聊天诉求**： 有些话题不属于特定助手，需要一个 general 的助手来承接，这也是我们为什么起名叫「随便聊聊」的原因，并且写死了助手的名称与头像。

所以依据这个设定，其实我们一开始并没有打算放开随便聊聊的助手设定（某一旧版本的界面如下）：
<img width="1290" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/5cc880a3-e34f-4385-8878-ebae2613d565">

然后在后续的迭代使用中，用户提出了对默认助手的修改诉求（#538），考虑到便捷性的问题，才放开了随便聊聊的设置入口，便于一键直达默认助手的设置页。

------

接下来讨论下你提到的几个点。

> ”随便聊聊“的右上角菜单应该同样导向“会话设置”
> “默认助手”改为“新建助手模板”（或者其他更突出“模板”的名字）

从产品上来说，我是希望用户能 GET 到配置随便聊聊，就是配置默认助手的参数（除了名称描述），这样的好处在于用户新建助手时，配置的创建是符合即时使用习惯的。如果分开成两个，会导致在随便聊聊改一次，然后未来在默认助手那里又要改一次，我自己觉得这个不是很自然。

现在的方案对于常用参数的配置来说更加友好。

> “随便聊聊”的“会话设置”应该可以正常更改名字和描述，并且 prompt 和 ”默认助手“ 独立


随便聊聊作为 LobeChat 的内置角色，我们不打算支持修改名称和描述，毕竟这相当于是一个应用内建的助手。（或者请告诉我为什么你需要修改它的名字？）


> 对于“新建助手”或者“新建对话”：

> 设置中提供开关来关闭列表右上方的按钮，或者提供选项设置按钮为”新建助手“或者”用随便聊聊进行对话“
> 更改右上角新建助手的图标，更加突出“助手”而不是“对话”

新建的逻辑你说的是合理的，未来我们做 Agent Builder 时，会进一步优化这块的设计。

-----

> 其他易用性建议（这似乎应该是另外的功能建议）：

> 提供一个更强大的检索界面（因为“助手-对话”有些时候比较难查找）

有计划：https://github.com/lobehub/lobe-chat/discussions/583

> 提供另外一种界面逻辑选项，类似openAI的官网界面（陈列来自所有助手的所有对话，新建对话时选择助手）

暂时不计划这么做

**Comment by samxiaowastaken** [comment]
我一开始不是很想改这个默认助手的名字，但是我在尝试修改设置的时候被设置的逻辑搞麻了，搞了半天发现改不了……并且在探索之后发现

> 从其他助手的设置可以进入“随便聊聊”的真实设置界面“会话设置”，但是改名字和描述不会生效

（按照您的描述，这个情况不应存在，好像是一个界面错误）

我又回去看了#538，在发issue前我也看过这一篇，诉求和我这里的很相似。简而言之，现在的设置并没有直接的给用户反馈。用户点进“随便聊聊”的设置菜单，发现可以改名和描述，改了之后发现没有变化……这是很不合理的事情。并且所有助手都能改名，唯独这个默认的“随便聊聊”不能，也显得非常奇怪。而社区中的助手大多以功能为名字（例如”托福写作辅导“、”翻译专家“、”产品文案撰写“），”随便聊聊“在这里也显得非常不合群。以及其他的系统名字“默认列表”“默认话题”也让这名字显得很怪。而默认助手的描述”开启大脑集群……“也并没有对助手的功能进行说明。

我认为英文的“chat randomly”是相对适合的，因为毕竟表达了可以谈论任何话题的意思，但是很显然“随便聊聊”在中文中的意思和这个不一样，给人一种这个助手办不了正事的感觉（）

虽然改一个更加正式一点的名字，然后设置界面上说明为什么不可更改，就可以解决，但是如果开放改名和描述的话可以直接省去这些问题。lobechat依然会有一个不可删除的默认助手，但是可以按照用户的喜好进行定义，同时设置的逻辑也更加清晰了。对于用户来说，这会是很大的便利。

> 从产品上来说，我是希望用户能 GET 到配置随便聊聊，就是配置默认助手的参数（除了名称描述），这样的好处在于用户新建助手时，配置的创建是符合即时使用习惯的。如果分开成两个，会导致在随便聊聊改一次，然后未来在默认助手那里又要改一次，我自己觉得这个不是很自然。

很明显用户没有完全明白这一点（不然我也不会来提这个问题，也不会有#538）。但是用户创建新助手时，也就代表着他有**不符合**即时使用习惯的需求，需要一个另外的助手来进行补充。此时这个默认助手（也就是模板）看起来也就是多此一举。实际上我在新建助手时的第一步就是把自动填充的内容全删了。至少在我和我所认识的几位同样适用lobechat的用户中，“默认助手”并没有发挥出模板的作用，反而是作为“随便聊聊”的一个**残缺的设置**来使用，并且从上文你也能看出，缺少了自定义名字和描述让这个设置菜单更加奇怪了。

> 随便聊聊作为LobeChat 内置的助手，我们在设计之初的目的有两个：
> 
> 快速引导用户上手： 新用户一上来就能用，不用再自己创建一个默认助手；
> 承接一些通用的聊天诉求： 有些话题不属于特定助手，需要一个 general 的助手来承接，这也是我们为什么起名叫「随便聊聊」的原因，并且写死了助手的名称与头像。

我现在理解了你们设置这个助手的目的，但是这个助手在后期的使用中，连同设置中的“默认助手”，并没有带来好的体验。那既然需要引导新用户上手，并且保证始终有一个助手来承接， 那不如将现在的“随便聊聊”换成“默认聊天”，描述留空，但是提供prompt设置，然后在助手中提供一个叫做“使用向导”的助手，并且包含prompt来解决用户使用lobechat中的问题（我记得github有免费issue查询API的来着，可以用那玩意），这样还能减轻开发者回答问题的压力。

---
最后感谢此项目开发者，我真的很喜欢lobechat的对话功能，可以删除和更改聊天记录的功能非常好用，包括token显示等。毫不夸张地说，这是我用过体验最好的界面了。

**Comment by woodytang** [comment]
首先感谢lobe-chat提供的非凡体验
看了上面2位的讨论，对于这个问题，个人感觉产品交互逻辑是有点问题的
如果随便聊聊 的有些项目 是固定 不可更改的，那么设置它应该在另一个地方，不要违反职责单一原则

---建议修改---
设置-> 默认助手 修改为 设置-> 新建助手默认设置（或参数/配置）
这个的作用就是可以快速开始启动新助手，就是一个模版，不进行任何实际助手实例的配置
然后无论是手动新建的助手，还是默认的随便聊聊，需要修改设置，都进入自己对应的设置页面
其中随便聊聊 的设置中，助手信息是不开放编辑的

**Reply by arvinxx** [reply]
你的思路没错。后续是准备这么做

**Comment by Moeblack** [comment]
随便聊聊的问题是部分API需要必须设置max_token才能正常使用
随便聊聊不能设置这个参数
发现我的issue可以关闭了

**Comment by arvinxx** [comment]
2025.02.18 callback 一下，目前 db 版本和 pglite 已经这么做了，等正式发布 V2 就可以全量获得该体验优化
